ee/hogai/eval/ci/eval_memory_onboarding.py

```diff
@@ -0,0 +1,397 @@
+from typing import Optional
+from uuid import uuid4
+
+import pytest
+
+from autoevals.llm import LLMClassifier
+from braintrust import EvalCase
+from braintrust_core import Score
+
+from posthog.schema import AssistantMessage, EventTaxonomyItem, HumanMessage
+
+from posthog.models.user import User
+from posthog.sync import database_sync_to_async
+
+from ee.hogai.graph import AssistantGraph
+from ee.hogai.graph.memory.prompts import ENQUIRY_INITIAL_MESSAGE, SCRAPING_TERMINATION_MESSAGE
+from ee.hogai.graph.root.nodes import SLASH_COMMAND_INIT
+from ee.hogai.utils.types import AssistantNodeName, AssistantState
+from ee.models.assistant import Conversation
+
+from ..base import MaxPublicEval
+
+
+class MemoryLLMClassifier(LLMClassifier):
+    def _run_eval_sync(self, output, expected, **kwargs):
+        if not output:
+            return Score(name=self._name(), score=None)
+        if expected in [ENQUIRY_INITIAL_MESSAGE, SCRAPING_TERMINATION_MESSAGE]:
+            # For the special cases, we MUST see the expected messages verbatim
+            return Score(
+                name=self._name(),
+                score=1.0 if isinstance(output, AssistantMessage) and output.content == expected else 0.0,
+            )
+        return super()._run_eval_sync(output, expected, **kwargs)
+
+    async def _run_eval_async(self, output, expected, **kwargs):
+        if not output:
+            return Score(name=self._name(), score=None)
+        if expected in [ENQUIRY_INITIAL_MESSAGE, SCRAPING_TERMINATION_MESSAGE]:
```

Comment:
Nitpick: Seems like you check for the exact match, but in the code it's `SCRAPING_TERMINATION_MESSAGE in answer`, so not 100%.